src/client/src/apps/MainRoute/widgets/footer-version/FooterVersion.tsx

```diff
@@ -1,12 +1,51 @@
-import React, { FC } from 'react'
+import React, { FC, useState } from 'react'
 import styled from 'styled-components/macro'
 
 export const Wrapper = styled.div`
   color: ${props => props.theme.textColor};
   opacity: 0.59;
   font-size: 0.75rem;
 `
+export const Link = styled.a`
+  color: inherit;
+  text-decoration: inherit;
+`
+const FooterVersion: FC = () => {
+  const [versionExists, setVersionExists] = useState<boolean | void>(false)
+  const gitTagExists = async (gitTag: string | undefined) => {
```

Comment:
I'm not sure embedding and calling this function inside the React component is a great idea. It will probably re-run on render.

What I would suggest is (1) extract the code outside of the react component into a named function (2) call it from within the component if you need, but if you do so, then use a useEffect hook.

However, is it necessary to call this inside the component? Could we do it once when the application starts up? The version is not likely to change once the application is launched!

Reply:
Would you recommend defining the gitTagExists async function outside of FooterVersion and then using a useEffect to get it's value on mount of the FooterVersion component? I did actually attempt to do this earlier but I kept running into this issue that the async function can only return a Promise whereas ideally I'd like it to simply return a boolean (true or false depending on whether the version number exists in the releases) and use that boolean to determine whether to display a clickable link or not. If I were to define the gitTagExists outside of my functional component, how would I then return a boolean true or false from it to use in my functional component? Currently I simply set State within the gitTagExists function but I agree that that isn't the optimal solution.